File: apps/web/src/api/legacy-import-list8.ts
import client from './client'

export interface LegacyImportList8VO {
  id: number
  code: string
  manager: string
  destination: string
  tax: string
  material: string
  kensa: string
  num: number
  pieces: number
  weight: number
  weight2: number
  length: number
  location: string
  date1: string
  status: string
  other: string
  unitCh: number
  rate: number
  souko: string
  factoryAddr: string
  updatetime: string
  updateuser: string
  showFlag: number
  rireki: string
}

export interface LegacyImportList8PageResponse {
  content: LegacyImportList8VO[]
  totalElements: number
  totalPages: number
  pageNumber: number
}

export interface LegacyImportList8UpdateCmd {
  code?: string
  manager?: string
  destination?: string
  tax?: string
  material?: string
  kensa?: string
  num?: number
  pieces?: number
  weight?: number
  weight2?: number
  length?: number
  location?: string
  date1?: string
  status?: string
  other?: string
  unitCh?: number
  rate?: number
  souko?: string
  factoryAddr?: string
  showFlag?: number
  rireki?: string
}

export const legacyImportList8Api = {
  list(params: {
    page?: number
    pageSize?: number
    code?: string
    location?: string
    souko?: string
    destination?: string
  }) {
    return client.get<LegacyImportList8PageResponse>('/legacy-import-list8', { params })
  },
  get(id: number) {
    return client.get<LegacyImportList8VO>(`/legacy-import-list8/${id}`)
  },
  create(data: LegacyImportList8UpdateCmd) {
    return client.post<LegacyImportList8VO>('/legacy-import-list8', data)
  },
  update(id: number, data: LegacyImportList8UpdateCmd) {
    return client.put<LegacyImportList8VO>(`/legacy-import-list8/${id}`, data)
  },
  delete(id: number) {
    return client.delete(`/legacy-import-list8/${id}`)
  },
  count() {
    return client.get<number>('/legacy-import-list8/count')
  },
}

File: apps/manpou-allinone/src/main/java/com/manpou/allinone/legacyimportlist8/domain/model/LegacyImportList8.java
package com.manpou.allinone.legacyimportlist8.domain.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "legacy_import_list8")
@Access(AccessType.FIELD)
@Getter
@Setter
public class LegacyImportList8 {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;

    @Column(name = "code", columnDefinition = "text")
    private String code;

    @Column(name = "manager", columnDefinition = "text")
    private String manager;

    @Column(name = "destination", columnDefinition = "text")
    private String destination;

    @Column(name = "tax", columnDefinition = "text")
    private String tax;

    @Column(name = "material", columnDefinition = "text")
    private String material;

    @Column(name = "kensa")
    private String kensa;

    @Column(name = "num")
    private Integer num;

    @Column(name = "pieces")
    private Integer pieces;

    @Column(name = "weight")
    private Double weight;

    @Column(name = "weight2")
    private Double weight2;

    @Column(name = "length")
    private Double length;

    @Column(name = "location", columnDefinition = "text")
    private String location;

    @Column(name = "date1")
    private LocalDate date1;

    @Column(name = "status", columnDefinition = "text")
    private String status;

    @Column(name = "other", columnDefinition = "text")
    private String other;

    @Column(name = "unit_ch")
    private Double unitCh;

    @Column(name = "rate")
    private Double rate;

    @Column(name = "souko", columnDefinition = "text")
    private String souko;

    @Column(name = "factory_addr")
    private String factoryAddr;

    @Column(name = "updatetime")
    private LocalDateTime updatetime;

    @Column(name = "updateuser", columnDefinition = "text")
    private String updateuser;

    @Column(name = "showFlag")
    private Integer showFlag;

    @Column(name = "rireki", columnDefinition = "longtext")
    private String rireki;
}
